net: ipv4: Fix ARP probe check in address conflict detection

anhuba opened the pull request:
The second condition needs to check ARP probes only

Fixes #80429

jukkar commented on subsys/net/ip/ipv4_acd.c:
```diff
@@ -288,15 +288,19 @@ enum net_verdict net_ipv4_acd_input(struct net_if *iface, struct net_pkt *pkt)
 					  (uint8_t *)&ifaddr->address.in_addr) &&
-		     memcmp(&arp_hdr->src_hwaddr, ll_addr->addr, ll_addr->len) != 0)) {
+		     (memcmp(&arp_hdr->src_hwaddr, ll_addr->addr, ll_addr->len) != 0) &&
+		     (net_ipv4_addr_cmp_raw(arp_hdr->src_ipaddr,
+		                           (uint8_t *)&all_zero_ip)))) {
```
We could avoid extra variable by this
```
(net_ipv4_addr_cmp_raw(arp_hdr->src_ipaddr,
	               (uint8_t *)(&(struct in_addr)INADDR_ANY_INIT)))) {
```

jukkar commented:
Could you also add text from your issue to the commit message, at least this could be put there
```
The ACD is not properly implemented as described in RFC5227 ch. 2.1.1
The implementation incorrectly detects an IP conflict, if an ARP request is received for the target IP.
The reason is that the current implementation checks for ARP requests instead of ARP probes.
```

rlubos reviewed (commented):
The fix itself looks good, but the branch needs some cleanup:
* Please use rebase instead of merge, no merge commits allowed
* The commit message in the fix commit is missing signoff entry.

anhuba commented:
changed commit message and elminated local variable

Commit:
net: ipv4: Fix ARP probe check in address conflict detection

The second condition needs to check ARP probes only

The ACD is not properly implemented as described in RFC5227 ch. 2.1.1
The implementation incorrectly detects an IP conflict, if an ARP request
is received for the target IP.
The reason is that the current implementation checks for ARP requests
instead of ARP probes.

Signed-off-by: Andreas Huber <andreas.huber@ch.sauter-bc.com>
